apiserver/utility/utility.go

```diff
@@ -27,61 +26,34 @@ func ValidateBitriseConfigAndSecret(bitriseConfig, secretsConfig string) (*Warni
 	secretsConfigBase64 := base64.StdEncoding.EncodeToString([]byte(secretsConfig))
 
-	validateCmd := command.New("bitrise",
-		"-l=panic", "validate", "--format=json",
-		"--config-base64", bitriseConfigBase64,
-		"--inventory-base64", secretsConfigBase64)
-
-	combinedOut, cmdErr := validateCmd.RunAndReturnTrimmedCombinedOutput()
-
-	type BitriseCLIValidateItemModel struct {
-		IsValid  bool     `json:"is_valid"`
-		Error    string   `json:"error"`
-		Warnings []string `json:"warnings"`
-	}
-	type BitriseCLIValidateDataModel struct {
-		Config  BitriseCLIValidateItemModel `json:"config"`
-		Secrets BitriseCLIValidateItemModel `json:"secrets"`
-	}
-	type BitriseCLIValidateOutputModel struct {
-		Data  BitriseCLIValidateDataModel `json:"data"`
-		Error string                      `json:"error"`
-	}
-
-	var validationOutput BitriseCLIValidateOutputModel
-	if outputParseErr := json.NewDecoder(strings.NewReader(combinedOut)).Decode(&validationOutput); outputParseErr != nil {
-		if cmdErr != nil {
-			return nil, fmt.Errorf("Failed to run bitrise validate command, error: %s | 'bitrise validate' command output: %s", cmdErr, combinedOut)
-		}
-		return nil, fmt.Errorf("Failed to parse bitrise validate output, error: %s | 'bitrise validate' command output: %s", outputParseErr, combinedOut)
+	_, bitriseWarns, bitriseErr := cli.CreateBitriseConfigFromCLIParams(bitriseConfigBase64, "")
+	var secretsErr error
+	if secretsConfigBase64 != "" {
+		_, secretsErr = cli.CreateInventoryFromCLIParams(secretsConfigBase64, "")
 	}
 
 	errorStrs := []string{}
-	if !validationOutput.Data.Config.IsValid {
-		errorStrs = append(errorStrs, "Config validation error: "+validationOutput.Data.Config.Error)
+	if bitriseErr != nil {
+		errorStrs = append(errorStrs, "Config validation error: "+bitriseErr.Error())
 	}
-	if !validationOutput.Data.Secrets.IsValid {
-		errorStrs = append(errorStrs, "Secret validation error: "+validationOutput.Data.Secrets.Error)
+	if secretsErr != nil {
+		errorStrs = append(errorStrs, "Secret validation error: "+secretsErr.Error())
+	} else {
+		if secretsConfigBase64 == "" {
+			errorStrs = append(errorStrs, "Secret validation error: Validation failed: Empty secrets configuration")
+		}
 	}
 
 	if len(errorStrs) > 0 {
 		return nil, fmt.Errorf("Validation failed: %s", strings.Join(errorStrs, " | "))
 	}
 
-	if len(validationOutput.Error) > 0 {
-		return nil, fmt.Errorf("bitrise validation command failed, error: %s", validationOutput.Error)
-	}
-
-	if cmdErr != nil {
-		return nil, fmt.Errorf("bitrise validation command failed, error: %s", cmdErr)
-	}
-
 	warningItems := &WarningItems{Config: []string{}, Secrets: []string{}}
-	if len(validationOutput.Data.Config.Warnings) > 0 {
-		warningItems.Config = validationOutput.Data.Config.Warnings
+	if len(bitriseWarns) > 0 {
+		warningItems.Config = bitriseWarns
 	}
-	if len(validationOutput.Data.Config.Warnings) > 0 {
-		warningItems.Secrets = validationOutput.Data.Secrets.Warnings
+	if secretsErr != nil {
```

Comment:
you can remove this case: secret validation will never raise warnings.
other than that, the secret error is stored in `errorStrs` already.